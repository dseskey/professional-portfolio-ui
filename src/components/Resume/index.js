import React from 'react';
import { Container, Row, Col, Card, ListGroup } from 'react-bootstrap';
import pdfResume from '../../assets/pdf/SeskeyResume.pdf';
import skillsJson from '../../assets/json/skills.json';

function Resume() {
    const tableStyle = { width: '18rem', marginTop: '1em', marginBottom: '1em', textAlign: 'center', background: 'grey', color: 'black' };
    return (
        <Container fluid data-testid='resume-container' id='resume-container' className='pagesParentContainer'>
            <Row className="pageHeader">
                <h2 data-testid='resume-section-header'>My Resume</h2>
            </Row>
            <Row id='resume-section-description'>
                The tables below display various technologies, infrastructures, etc that I have worked with. You can also download a PDF version of my <a href={pdfResume} rel="noreferrer"  target="_blank"> resume</a>.
            </Row>
            <Row id='resume-proficiencies-section' data-testid='resume-proficiencies-section'>
                {skillsJson.topics.map((topic) => {
                    return (
                        <Col data-testid={`${topic.title}-table`} key={topic.title+'-table-col'} className='resumeSkillTable'>
                            <Card key={topic.title+'-card'}style={tableStyle}>
                                <Card.Header key={topic.title+'-card-header'} className='skillTableHeader'>{topic.title}</Card.Header>
                                <ListGroup data-testid={`${topic.title}-list-group`} key={topic.title+'-list-group'} variant="flush">
                                    {topic.skills.map((skill) => {
                                        return (
                                            <ListGroup.Item key={topic+'-'+skill+'-list-item'}>{skill}</ListGroup.Item>
                                        )
                                    })
                                    }
                                </ListGroup>
                            </Card>
                        </Col>
                    )
                })}
            </Row>

        </Container>
    )
}

export default Resume;