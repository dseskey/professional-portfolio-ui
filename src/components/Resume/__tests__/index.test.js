import React from 'react';
import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Resume from '..'

afterEach(cleanup);



describe("Resume component", () =>{
    it('renders', () => {
        render(<Resume/>);
    });

    it('matches snapshot DOM node structure', () => {
        const {asFragment} = render(<Resume/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('has a Resume section with the proper shared class name for styling (pagesParentContainer)', () => {
        const { getByTestId } = render(<Resume/>);
        expect(getByTestId('resume-container')).toHaveClass('pagesParentContainer');
        expect(getByTestId('resume-container')).toHaveAttribute('id','resume-container')
    });

    it('has a section header of My Resume', () => {
        const { getByTestId } = render(<Resume/>);
        expect(getByTestId('resume-section-header')).toHaveTextContent('My Resume');
    });

    it('has an Architecture table of skills based on the data load with proper class styling (resumeSkillTable)', () => {
        const { getByTestId } = render(<Resume/>);
        expect(getByTestId('Architecture-table')).toHaveClass('resumeSkillTable');
    });

    it('has a resume proficincies section with 4 children based on the dynamic data rendering.', () => {
        const { getByTestId } = render(<Resume/>);
        expect(getByTestId('resume-proficiencies-section')).toHaveClass('row');
        expect(getByTestId('resume-proficiencies-section').childElementCount).toEqual(4);
    });

    it('has a FrontEnd Development Listgroup with 9 skills', () => {
        const { getByTestId } = render(<Resume/>);
        expect(getByTestId('Frontend Development-list-group')).toHaveClass('list-group');
        expect(getByTestId('Frontend Development-list-group').childElementCount).toEqual(9);
    });



});






/*
 (
        <Container fluid data-testid='resume-container' id='resume-container' className='pagesParentContainer'>
            <Row className="pageHeader">
                <h2 data-testid='resume-section-header'>My Resume</h2>

                Architecture-table
*/