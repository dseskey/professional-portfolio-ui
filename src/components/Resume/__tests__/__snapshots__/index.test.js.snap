// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resume component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="pagesParentContainer container-fluid"
    data-testid="resume-container"
    id="resume-container"
  >
    <div
      class="pageHeader row"
    >
      <h2
        data-testid="resume-section-header"
      >
        My Resume
      </h2>
    </div>
    <div
      class="row"
      id="resume-section-description"
    >
      The tables below display various technologies, infrastructures, etc that I have worked with. You can also download a PDF version of my 
      <a
        href="SeskeyResume.pdf"
        rel="noreferrer"
        target="_blank"
      >
         resume
      </a>
      .
    </div>
    <div
      class="row"
      data-testid="resume-proficiencies-section"
      id="resume-proficiencies-section"
    >
      <div
        class="resumeSkillTable col"
        data-testid="Architecture-table"
      >
        <div
          class="card"
          style="width: 18rem; margin-top: 1em; margin-bottom: 1em; text-align: center; background: grey; color: black;"
        >
          <div
            class="skillTableHeader card-header"
          >
            Architecture
          </div>
          <div
            class="list-group list-group-flush"
            data-testid="Architecture-list-group"
          >
            <div
              class="list-group-item"
            >
              Software Architecture Professional Certificate (CMU SEI)
            </div>
            <div
              class="list-group-item"
            >
              Three Tier Architecture
            </div>
            <div
              class="list-group-item"
            >
              AWS Architecture
            </div>
            <div
              class="list-group-item"
            >
              Event Driven Architecture
            </div>
            <div
              class="list-group-item"
            >
              Mobile Architecture
            </div>
            <div
              class="list-group-item"
            >
              Data Architecture
            </div>
            <div
              class="list-group-item"
            >
              Security Architecture
            </div>
          </div>
        </div>
      </div>
      <div
        class="resumeSkillTable col"
        data-testid="Frontend Development-table"
      >
        <div
          class="card"
          style="width: 18rem; margin-top: 1em; margin-bottom: 1em; text-align: center; background: grey; color: black;"
        >
          <div
            class="skillTableHeader card-header"
          >
            Frontend Development
          </div>
          <div
            class="list-group list-group-flush"
            data-testid="Frontend Development-list-group"
          >
            <div
              class="list-group-item"
            >
              React
            </div>
            <div
              class="list-group-item"
            >
              React Native
            </div>
            <div
              class="list-group-item"
            >
              HTML
            </div>
            <div
              class="list-group-item"
            >
              CSS
            </div>
            <div
              class="list-group-item"
            >
              jQuery
            </div>
            <div
              class="list-group-item"
            >
              JavaScript
            </div>
            <div
              class="list-group-item"
            >
              Vue
            </div>
            <div
              class="list-group-item"
            >
              JSP
            </div>
            <div
              class="list-group-item"
            >
              Angular
            </div>
          </div>
        </div>
      </div>
      <div
        class="resumeSkillTable col"
        data-testid="Backend Development-table"
      >
        <div
          class="card"
          style="width: 18rem; margin-top: 1em; margin-bottom: 1em; text-align: center; background: grey; color: black;"
        >
          <div
            class="skillTableHeader card-header"
          >
            Backend Development
          </div>
          <div
            class="list-group list-group-flush"
            data-testid="Backend Development-list-group"
          >
            <div
              class="list-group-item"
            >
              Node.js
            </div>
            <div
              class="list-group-item"
            >
              Java
            </div>
            <div
              class="list-group-item"
            >
              C#
            </div>
            <div
              class="list-group-item"
            >
              CSS
            </div>
            <div
              class="list-group-item"
            >
              jQuery
            </div>
            <div
              class="list-group-item"
            >
              Vue
            </div>
            <div
              class="list-group-item"
            >
              JSP
            </div>
            <div
              class="list-group-item"
            >
              Angular
            </div>
          </div>
        </div>
      </div>
      <div
        class="resumeSkillTable col"
        data-testid="Infrastructure Databases-table"
      >
        <div
          class="card"
          style="width: 18rem; margin-top: 1em; margin-bottom: 1em; text-align: center; background: grey; color: black;"
        >
          <div
            class="skillTableHeader card-header"
          >
            Infrastructure Databases
          </div>
          <div
            class="list-group list-group-flush"
            data-testid="Infrastructure Databases-list-group"
          >
            <div
              class="list-group-item"
            >
              AWS
            </div>
            <div
              class="list-group-item"
            >
              Pivotal Cloud Foundry
            </div>
            <div
              class="list-group-item"
            >
              MuleSoft
            </div>
            <div
              class="list-group-item"
            >
              OpenShift
            </div>
            <div
              class="list-group-item"
            >
              MongoDB
            </div>
            <div
              class="list-group-item"
            >
              Cassandra
            </div>
            <div
              class="list-group-item"
            >
              SQL
            </div>
            <div
              class="list-group-item"
            >
              MySQL
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
